### PR TITLE
fix: password read on windows

### PR DESCRIPTION
**Pull request by matejvasek**
Signed-off-by: Matej Vasek <mvasek@redhat.com>

### CODE REVIEW

**Comment by matejvasek** [comment]
Beware of the above code; I didn't tested it.

File: cmd/deploy.go [hunk 1]
```diff
@@ -200,7 +201,7 @@ func getPassword(ctx context.Context) ([]byte, error) {
 	})
 
 	go func() {
-		pass, err := term.ReadPassword(0)
+		pass, err := term.ReadPassword(int(syscall.Stdin)) // nolint: unconvert
```
Comment (lkingland) [inline cmd/deploy.go:204]: Good to know!

**Review by lkingland — APPROVED** [review]

**Comment by matejvasek** [comment]
@lkingland I don't even know how to write integration test for this.

First idea might be `echo  "mvasek\npasswordhere\n" | f deploy --build=false -v`. But that won't work.
It does causes `read` syscall to get expected input. However for password you also need to invoke `ioctl` syscall to disable character echoing. result is `inappropriate ioctl for device`. :disappointed:

**Comment by matejvasek** [comment]
@ramo

> @lkingland I don't even know how to write integration test for this.
> 
> First idea might be `echo "mvasek\npasswordhere\n" | f deploy --build=false -v`. But that won't work.
> It does causes `read` syscall to get expected input. However for password you also need to invoke `ioctl` syscall to disable character echoing. result is `inappropriate ioctl for device`. 
> 
> EDIT:
> it can be done by `expect -c 'spawn -noecho func deploy --build=false -v; send "mvasek\npwdhere\n"; interact'`

@jrangelramos it might be good to test this in our e2e tests if possible.

**Review by lance — APPROVED** [review]

**Comment by matejvasek** [comment]
@boson-project/core I added just tiny amendment: trimming of `\r` character.

**Comment by jrangelramos** [comment]
Yea. @matejvasek That's a good point to test and ensure it won't break on future.  I think using `expect` as you mentioned could work as well

**Pull request merged**